server/src/main/java/io/druid/server/lookup/cache/LookupCoordinatorManager.java

```diff
@@ -506,137 +274,426 @@ public boolean deleteLookup(final String tier, final String lookup, AuditInfo au
+              {
+                lookupMgmtLoop();
               }
-              if (allLookupTiers == null) {
-                LOG.info("Not updating lookups because no data exists");
-                return;
+            },
+            2000,
+            lookupCoordinatorManagerConfig.getPeriod(),
+            TimeUnit.MILLISECONDS
+        );
+        Futures.addCallback(
+            backgroundManagerFuture, new FutureCallback()
+            {
+              @Override
+              public void onSuccess(@Nullable Object result)
+              {
+                backgroundManagerExitedLatch.countDown();
+                LOG.debug("Exited background lookup manager");
               }
-              for (final String tier : allLookupTiers.keySet()) {
-                try {
-                  final Map<String, Map<String, Object>> allLookups = allLookupTiers.get(tier);
-                  final Map<String, Map<String, Object>> oldLookups = prior_update.get(tier);
-                  final Collection<String> drops;
-                  if (oldLookups == null) {
-                    drops = ImmutableList.of();
-                  } else {
-                    drops = Sets.difference(oldLookups.keySet(), allLookups.keySet());
-                  }
-                  if (allLookupTiers == prior_update) {
-                    LOG.debug("No updates");
-                    updateAllNewOnTier(tier, allLookups);
-                  } else {
-                    updateAllOnTier(tier, allLookups);
-                    deleteAllOnTier(tier, drops);
-                  }
-                }
-                catch (InterruptedException e) {
-                  Thread.currentThread().interrupt();
-                  throw Throwables.propagate(e);
-                }
-                catch (Exception e) {
-                  LOG.error(e, "Error updating lookups for tier [%s]. Will try again soon", tier);
+
+              @Override
+              public void onFailure(Throwable t)
+              {
+                backgroundManagerExitedLatch.countDown();
+                if (backgroundManagerFuture.isCancelled()) {
+                  LOG.debug("Exited background lookup manager due to cancellation.");
+                } else {
+                  LOG.makeAlert(t, "Background lookup manager exited with error!").emit();
                 }
               }
-              prior_update = allLookupTiers;
             }
-          },
-          0,
-          lookupCoordinatorManagerConfig.getPeriod(),
-          TimeUnit.MILLISECONDS
-      );
-      Futures.addCallback(backgroundManagerFuture, new FutureCallback<Object>()
-      {
-        @Override
-        public void onSuccess(@Nullable Object result)
-        {
-          backgroundManagerExitedLatch.countDown();
-          LOG.debug("Exited background lookup manager");
+        );
+
+        LOG.debug("Started");
+      }
+      catch (Exception ex) {
+        LOG.makeAlert(ex, "Got Exception while start()").emit();
```

Comment:
getting here is unexpected, we reach here if executorService from last run hasn't finished yet, which is checked first... or something terrible has gone wrong ...  so alert is generated.... lookups are not managed at this state and node restart is necessary .... or on next stop() and then start() again, it will try to recover (which might be a long while)